packages/core/test/profiles-determinism.test.ts

```
@@ -0,0 +1,85 @@
+    const b = prepareChaosConfig({ profile: 'mobile-checkout', seed: 1 });
+    expect(JSON.stringify(a)).toBe(JSON.stringify(b));
+  });
+
+  it('profile-only config produces same rule event ordering across calls', () => {
+    const a = prepareChaosConfig({ profile: 'mobile-checkout', seed: 99 });
+    const b = prepareChaosConfig({ profile: 'mobile-checkout', seed: 99 });
+    // Rule arrays must be in identical order — that determines event sequence under PRNG.
+    expect(a.network!.latencies!.map((l) => l.urlPattern)).toEqual(b.network!.latencies!.map((l) => l.urlPattern));
+    if (a.network!.failures) {
+      expect(a.network!.failures.map((f) => `${f.urlPattern}:${f.statusCode}`))
+        .toEqual(b.network!.failures!.map((f) => `${f.urlPattern}:${f.statusCode}`));
```

ESLint

[error] Avoid calling `expect` inside conditional statements

(vitest/no-conditional-expect)